File: common/src/main/java/com/ssblur/scriptor/word/action/potions/WildPhasingAction.java
package com.ssblur.scriptor.word.action.potions;

import com.ssblur.scriptor.api.word.Descriptor;
import com.ssblur.scriptor.blockentity.PhasedBlockBlockEntity;
import com.ssblur.scriptor.effect.ScriptorEffects;
import com.ssblur.scriptor.helpers.targetable.Targetable;

public class WildPhasingAction extends PotionAction {
  public WildPhasingAction() {
    super(ScriptorEffects.WILD_PHASING.get(), 20, 1d/3d, new Cost(15, COSTTYPE.ADDITIVE));
  }

  @Override
  public void applyToPosition(Targetable caster, Targetable targetable, Descriptor[] descriptors, double strength, double duration) {
    var level = targetable.getLevel();
    var pos = targetable.getOffsetBlockPos();

    PhasedBlockBlockEntity.phase(level, pos);
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/action/potions/WildPhasingAction.kt
package com.ssblur.scriptor.word.action.potions

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.blockentity.PhasedBlockBlockEntity.Companion.phase
import com.ssblur.scriptor.effect.ScriptorEffects.WILD_PHASING
import com.ssblur.scriptor.helpers.targetable.Targetable

class WildPhasingAction: PotionAction(WILD_PHASING.ref(), 5.0, 1.0 / 3.0, Cost(15.0, COSTTYPE.ADDITIVE)) {
  override fun applyToPosition(
    caster: Targetable?,
    targetable: Targetable?,
    descriptors: Array<Descriptor>?,
    strength: Double,
    duration: Double
  ) {
    val level = targetable!!.level
    val pos = targetable.offsetBlockPos

    phase(level, pos)
  }
}
